tests/common/devices/sonic.py

```diff
@@ -530,10 +581,50 @@ def all_critical_process_status(self):
         """
         @summary: Check whether all critical processes status for all critical services
         """
-        result = {}
+        # Get critical process definition of all services
+        group_process_results = self.critical_group_process()
+
+        # Get process status of all services. Run cmds in batch to save overhead
+        cmds = []
         for service in self.critical_services:
-            result[service] = self.critical_process_status(service)
-        return result
+            cmd = 'docker exec {} supervisorctl status'.format(service)
+            cmds.append(cmd)
+        results = self.shell_cmds(cmds=cmds, continue_on_fail=True, module_ignore_errors=True)['results']
+
+        # Extract service name of each command result, transform results list to a dict keyed by service name
+        service_results = {}
+        for res in results:
+            service = res['cmd'].split()[2]
+            service_results[service] = res
+
+        # Parse critical process status of all services
+        all_critical_process = {}
+        for service in self.critical_services:
+            service_critical_process = {
+                'status': True,
+                'exited_critical_process': [],
+                'running_critical_process': []
+            }
+            if service not in group_process_results or service not in service_results:
+                service_critical_process['status'] = False
```

Comment:
If the service is not in `group_process_results`, ie `/etc/supervisor/critical_processes`, why should we consider it as `Failed`? I think it means the service is not critical.

Reply:
Variable `group_process_results` is a dict storing groups and processes for all critical services only. Each critical service should have an entry in the dict. If a critical service does not have an entry in it, it means something is wrong.